### PR TITLE
Dala 3300 remove pending datasets from qa if deleted

### PR DESCRIPTION
**Pull request by stewal19**
# Pull Request \<Title>
`<Description here>`
## Things to do during Peer Review
Please check all boxes before the Pull Request is merged. In case you skip a box, describe in the PRs description (that means: here) why the check is skipped.
- [x] The Github Actions (including Sonarqube Gateway and Lint Checks) are green. This is enforced by Github. 
- [ ] A peer-review has been executed
  - [ ] The code has been manually inspected by someone who did not implement the feature
  - [ ] If this PR includes work on the frontend, at least one `@ts-nocheck` is removed. Additionally, there should not be any `@ts-nocheck` in files modified by this PR. If no `@ts-nocheck` are left: Celebrate :tada: :confetti_ball: type-safety and remove this entry. 
- [ ] The PR actually implements what is described in the JIRA-Issue
- [ ] At least one test exists testing the new feature
  - [ ] If you have created new test files, make sure that they are included in a test container and actually run in the CI
- [ ] At least 3 consecutive CI runs are successfully executed to ensure that there are no flaky tests.
- [ ] Documentation is updated as required
- [ ] The automated deployment is updated if required
- [ ] If there was a database entity class added, there must also be a migration script for creating the corresponding database if flyway is already used by the service
- [ ] IF there are changes done to the framework data models or to a database entity class, the following steps were completed in order
  - [ ] A fresh clone of dataland.com is generated (see Wiki page on "OTC" for details)
  - [ ] The feature branch is deployed to clone with `Reset non-user related Docker Volumes & Re-populate` turned off
  - [ ] It's verified that the CD run is green
  - [ ] The new feature is manually used/tested/observed on the clone server. Testing of the new feature should (also) be done by a second, independent reviewer from the dev team
  - [ ] The feature branch is deployed to dev1 with `Reset non-user related Docker Volumes & Re-populate` turned on, and it's verified that the CD run is green  
- [ ] ELSE, the new version is deployed to the dev server "dev1" using this branch
  - [ ] Run with setting `Reset non-user related Docker Volumes & Re-populate` turned on 
  - [ ] It's verified that this version actually is the one deployed (check gitinfo for branch name and commit id!)
  - [ ] It's verified that the CD run is green
  - [ ] The new feature is manually used/tested/observed on dev server. Testing of the new feature should (also) be done by a second, independent reviewer from the dev team

### CODE REVIEW

File: dataland-qa-service/src/main/kotlin/org/dataland/datalandqaservice/services/QaReportEventListenerService.kt [hunk 1]
```diff
@@ -73,6 +77,8 @@ class QaReportEventListenerService(
         messageUtils.rejectMessageOnException {
             if (actionType == ActionType.DeleteData) {
                 reportManager.deleteAllQaReportsForDataId(dataId)
+                reviewQueueRepository.deleteByDataId(dataId)
+                reviewHistoryRepository.deleteByDataId(dataId)
```
Comment (emanueldfine) [inline dataland-qa-service/src/main/kotlin/org/dataland/datalandqaservice/services/QaReportEventListenerService.kt:81]: Hier war ich mir nicht ganz sicher:

Das ist ja eine history-Tabelle.
An einer history schraubt man ja normalerweise nicht rum, auch wenn es das betroffene dataset nicht mehr gibt.
Sollten wir die history nicht unangetastet lassen?

File: dataland-qa-service/src/main/kotlin/org/dataland/datalandqaservice/repositories/ReviewQueueRepository.kt [hunk 2]
```diff
@@ -16,4 +16,9 @@ interface ReviewQueueRepository : JpaRepository<ReviewQueueEntity, String> {
             "ORDER BY status.receptionTime ASC",
     )
     fun getSortedPendingDataIds(): List<String>
+
+    /**
+     * Deletes all queued QA requests for a specific dataId.
```
Comment (emanueldfine) [inline dataland-qa-service/src/main/kotlin/org/dataland/datalandqaservice/repositories/ReviewQueueRepository.kt:21]: Ist nicht nur ein einziger QA request in der queue?
Also wird nicht nur dieser eine einzige gelöscht?

Die Beschreibung klingt so, als könnten da auch mehrere sein.

File: dataland-qa-service/src/main/kotlin/org/dataland/datalandqaservice/repositories/ReviewHistoryRepository.kt [hunk 3]
```diff
@@ -6,4 +6,9 @@ import org.springframework.data.jpa.repository.JpaRepository
 /**
  * A JPA repository for accessing the historical review information of a dataset
  */
-interface ReviewHistoryRepository : JpaRepository<ReviewInformationEntity, String>
+interface ReviewHistoryRepository : JpaRepository<ReviewInformationEntity, String> {
+    /**
+     * Deletes all queued QA requests for a specific dataId.
+     */
+    fun deleteByDataId(dataId: String)
+}
```
Comment (emanueldfine) [inline dataland-qa-service/src/main/kotlin/org/dataland/datalandqaservice/repositories/ReviewHistoryRepository.kt:14]: Beschreibung stimmt nicht meines Erachtens.
Du löschst ja nur den history-Eintrag, oder.  Keine "queued QA requests".

Und auch hier:  Wieso löschen wir Dinge aus der history?  Eine history ist normalerweise unangetastet.

File: dataland-qa-service/src/main/kotlin/org/dataland/datalandqaservice/services/QaReportEventListenerService.kt (hunk 1 above)
Comment (stewal19) [reply dataland-qa-service/src/main/kotlin/org/dataland/datalandqaservice/services/QaReportEventListenerService.kt:81]: Fairer Punkt. Aber die QA Reports sind doch eigentlich auch eine Historie. Plus es stehen einfach Einträge dann in der Tabelle mit denen man nichts mehr anfangen kann, da ja auch alle meta informationen zu der dataId gelöscht werden.

File: dataland-qa-service/src/main/kotlin/org/dataland/datalandqaservice/repositories/ReviewQueueRepository.kt (hunk 2 above)
Comment (stewal19) [reply dataland-qa-service/src/main/kotlin/org/dataland/datalandqaservice/repositories/ReviewQueueRepository.kt]: Angepasst

File: dataland-qa-service/src/main/kotlin/org/dataland/datalandqaservice/repositories/ReviewHistoryRepository.kt (hunk 3 above)
Comment (stewal19) [reply dataland-qa-service/src/main/kotlin/org/dataland/datalandqaservice/repositories/ReviewHistoryRepository.kt]: Angepasst

**Pull request merged**